### ISSUE TITLE
Bug with getImageSnippetById ?

### ISSUE THREAD

**Issue by smoke2007** [annotorious/annotorious#217]
I've used following code :

```
console.log(annotationItem.annotation_data.id);
//output : #d902e6ff-d75b-40a3-9400-83ef4ab5e3a3

const {snippet, transform }= await anno.getImageSnippetById(annotationItem.annotation_data.id);
//Uncaught (in promise) TypeError: Cannot read properties of null (reading 'getBoundingClientRect')

//get the div with id : testcanvas
var testcanvas = document.getElementById('testcanvas');

//add the returned canvas element in the div as a child
testcanvas.appendChild(snippet);
```
 

I get this error : 
At first I thought it was due to using a formatter that changed the css classes, but i tried without the formatter and got the same result.
   
_Uncaught (in promise) TypeError: Cannot read properties of null (reading 'getBoundingClientRect')
    at af (ImageSnippet.js:16:29)
    at ZS.getImageSnippetById (OSDAnnotationLayer.js:435:14)
    at b_.getImageSnippetById (OpenSeadragonAnnotator.jsx:298:26)
    at S_.getImageSnippetById (index.jsx:175:23)
    at fetchAnnotations (x150060:450:45)_
   

which points to 
  
```
  export const getSnippet = (viewer, element) => {
  // Annotation shape could be the element itself or a 
  // child (in case of editable shapes, the element would be 
  // the group with shape + handles)
  const shape = hasClass(element, '.a9s-annotation') ? 
    element : element.querySelector('.a9s-annotation');

  // Scale factor for OSD canvas element (physical vs. logical resolution)
  const { canvas } = viewer.drawer;
  const canvasBounds = canvas.getBoundingClientRect();
  const kx = canvas.width / canvasBounds.width;
  const ky = canvas.height / canvasBounds.height;

  const shapeBounds = shape.getBoundingClientRect();

  const x = shapeBounds.x - canvasBounds.x;
  const y = shapeBounds.y - canvasBounds.y;
  const { width, height } = shapeBounds;

```

**Comment by rsimon** [comment]
Yikes, looking at this now: there are several issues with this method. First: there was a regression bug which broke the behavior in the past two (?) releases. This is [now fixed](https://github.com/recogito/annotorious-openseadragon/commit/a34aec602ceb427a8c3e86d50f801d14372c9747) and will go live with the next release, 2.7.4. It looks like your error stems exactly from this bug.

However there is one more issue: the way `getSnippet` works is that it creates a copy of the annotated area, as displayed in the OpenSeadragon viewer. This means __it is not possible to get a snippet for an annotation that is currently not visible on your screen.__ There is, unfortunately, no (easy) way around this. I'll update the docs accordingly. (Originally, `getSnippetById` was a community contribution, and I didn't think through the whole thing when I merged the pull request...)

Let me know if the most recent fix solves your error. If so, I'll close the issue once documentation is updated.

**Comment by smoke2007** [comment]
oh great I'll test it out with the new release. A pitty indeed about the visibility issue, still i'll see if I can get a use out of it. 
I'm building out an entire annotation workspace for interdisciplinary heritage science, so whatever I can add as extra's is good ;)

I was originally (2 years ago) planning to extend mirador 3 annotation plugin when I got to this point in my DAMS , but your software is leaps better and easy to customize.

**Comment by rsimon** [comment]
Just a quick update that I just published a new release (v2.7.5).

Thanks for the positive feedback & interesting ab about that DAMS you are developing. Is there any info public about it?

BTW: there's also a (very early...) integration of Annotatorious and Mirador 3, in case Mirador is still on your list.
